agent_secret_manager/src/lib.rs

```diff
@@ -1,50 +1,71 @@
-use agent_shared::config::{config, SecretManagerConfig};
-use did_manager::{InMemoryCache, SecretManager};
+use agent_shared::config::config;
+use did_manager_identity_stronghold_ext::StrongholdExtStorage;
+use identity_iota::{
+    storage::{JwkStorage, KeyId, KeyType},
+    verification::jws::JwsAlgorithm,
+};
+use iota_sdk::client::secret::stronghold::StrongholdSecretManager;
 use log::info;
 
 pub mod service;
 pub mod subject;
 
-// TODO: find better solution for this
-pub async fn secret_manager() -> SecretManager {
-    let SecretManagerConfig {
-        stronghold_path: snapshot_path,
-        stronghold_password: password,
-        issuer_eddsa_key_id,
-        issuer_es256_key_id,
-    } = config().secret_manager.clone();
+pub async fn stronghold_storage() -> StrongholdExtStorage {
+    #[cfg(feature = "test_utils")]
+    iota_stronghold::engine::snapshot::try_set_encrypt_work_factor(0).unwrap();
 
-    info!("{:?}", config().secret_manager);
+    info!("Initializing Stronghold storage");
 
-    let mut builder = SecretManager::builder()
-        .snapshot_path(&snapshot_path)
-        .password(&password);
+    let stronghold_password = config().secret_manager.stronghold_password.clone();
+    let stronghold_path = config().secret_manager.stronghold_path.clone();
 
-    if let Some(issuer_eddsa_key_id) = issuer_eddsa_key_id {
-        builder = builder.with_ed25519_key(&issuer_eddsa_key_id);
-    }
-
-    if let Some(issuer_es256_key_id) = issuer_es256_key_id {
-        builder = builder.with_es256_key(&issuer_es256_key_id);
-    }
+    info!("Stronghold path: {stronghold_path}");
 
-    if let Some(did_document_cache) = config().did_document_cache.clone() {
```

Comment:
Please consider removing the config entry as well if the cache is not used anymore.